DQM/SiStripCommon/src/TkHistoMap.cc

```diff
@@ -78,27 +85,43 @@ void TkHistoMap::loadTkHistoMap(std::string path, std::string MapName, bool mech
 }
 
 void TkHistoMap::createTkHistoMap(std::string& path, std::string& MapName, float& baseline, bool mechanicalView){
-  
+
   int nchX;
   int nchY;
   double lowX,highX;
   double lowY, highY;
   std::string fullName, folder;
 
-  tkHistoMap_.resize(HistoNumber);    
+  const bool bookTH2F = isTH2F_;
+  tkHistoMap_.resize(HistoNumber);
   for(int layer=1;layer<HistoNumber;++layer){
     folder=folderDefinition(path,MapName,layer,mechanicalView,fullName);
     tkdetmap_->getComponents(layer,nchX,lowX,highX,nchY,lowY,highY);
-    MonitorElement* me  = dqmStore_->bookProfile2D(fullName.c_str(),fullName.c_str(),
+    MonitorElement* me;
+
+    if(bookTH2F==false){
+
+    me  = dqmStore_->bookProfile2D(fullName.c_str(),fullName.c_str(),
```

Comment:
I checked the code by running runTheMatrix.py and it seems that the problematic constructor is called once, [here](https://github.com/cms-sw/cmssw/blob/master/DQM/SiStripMonitorHardware/src/FEDHistograms.cc#L810).

This can be fixed. Is this enough to make sure that the DQM will not crash?